applications/StructuralMechanicsApplication/custom_constitutive/plane_stress_d_plus_d_minus_damage_masonry_2d.cpp

```diff
@@ -0,0 +1,688 @@
+void DamageDPlusDMinusMasonry2DLaw::CalculateExponentialDamageTension(
+	const double UniaxialStress,
+	const double Threshold,
+	const double DamageParameter,
+	const double CharacteristicLength,
+	ConstitutiveLaw::Parameters& rValues,
+	double& rDamage)
+{
+	const Properties& r_material_prop = rValues.GetMaterialProperties();
+	const double initial_threshold = r_material_properties["YIELD_STRESS_TENSION"]
+	rDamage = 1.0 - (initial_threshold / UniaxialStress) * std::exp(DamageParameter * (1.0 - (UniaxialStress / initial_threshold)))
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+void DamageDPlusDMinusMasonry2DLaw::IntegrateStressVectorCompression(
+	array_1d<double,3>& rPredictiveStressVector,
+	const double UniaxialStress,
+    double& rDamage,
+    double& rThreshold,
+    ConstitutiveLaw::Parameters& rValues,
+    const double CharacteristicLength)
+{
+	DamageDPlusDMinusMasonry2DLaw::CalculateBezier3DamageCompression(UniaxialStress, rDamage, rThreshold, CharacteristicLength, rValues);
+	
+	rPredictiveStressVector *= (1.0 - rDamage);
+}
+/***********************************************************************************/
+/***********************************************************************************/
+void DamageDPlusDMinusMasonry2DLaw::CalculateBezier3DamageCompression(
+	const double UniaxialStress,
+	double& rDamage,
+	double& rThreshold,
+	const double CharacteristicLength,
+	ConstitutiveLaw::Parameters& rValues)
+{
+	// Call the Material Properties 
+	const Properties& r_material_prop = rValues.GetMaterialProperties();
+	const double E = r_material_properties[YOUNG_MODULUS]; 
+	const double s0 = r_material_properties["DAMAGE_ONSET_STRESS_COMPRESSION"];
+	const double sp = r_material_properties["YIELD_STRESS_COMPRESSION"];
+	const double ep = r_material_properties["YIELD_STRAIN_COMPRESSION"];
+	const double sr = r_material_properties["RESIDUAL_STRESS_COMPRESSION"];
+	const double c1 = r_material_properties["BEZIER_CONTROLLER_C1"];
+	const double c2 = r_material_properties["BEZIER_CONTROLLER_C2"];
+	const double c3 = r_material_properties["BEZIER_CONTROLLER_C3"];
+	const double Gc = r_material_properties["FRACTURE_ENERGY_COMPRESSION"];
+	
+	// Calculate missing Bezier Determinators
+	const double alpha = 2.0 * (ep - (sp/E));
+	const double e0 = s0 / E;
+	const double ei = sp / E;
+	const double sk = sr + (sp - sr) * c1;
+	const double ej = ep + alpha * c2;
+	const double ek = ej + alpha * (1.0 - c2);
+	const double er = ( (ek - ej) * (sp - sr)/(sp -sk) ) + ej;
+	const double eu = er * c3;
+	const double gc = Gc / CharacteristicLength;
+	
+	// Perform the Energy Regularization of the Bezier Determinators
+	double gc_bezier; 
+	DamageDPlusDMinusMasonry2DLaw::RegulateBezierDeterminators(gc, sp, sk, sr, ep, ej, ek, er, eu);
+	
+	// Compute rDamage
+	double StrainLikeCounterpart = UniaxialStress / E;

```

Comment:
const double

Reply:
and please follow the style guide => `strain_like_counterpart`